.github/workflows/build-and-test.yml

```diff
@@ -267,12 +267,24 @@ jobs:
     needs: [unittest, integration-tests, lint]
     strategy:
       matrix:
-        binary:
-          - darwin_amd64
-          - darwin_arm64
-          - linux_amd64
-          - linux_arm64
-          - windows_amd64
+        os:
+          - darwin
+          - linux
+          - windows
+        arch:
+          - 386
+          - amd64
+          - arm
+          - arm64
+        exclude:
+          - os: darwin
+            arch: 386
+          - os: darwin
+            arch: arm
+          - os: windows
+            arch: arm
+          - os: windows
+            arch: arm64
```

Comment:
I know this combo is listed in the [goreleaser](https://github.com/open-telemetry/opentelemetry-collector-releases/blob/49cc73c55020cc3b0a9db9b4aea992037b8037f2/.goreleaser.yaml#L12-L16) mentioned by @mx-psi in the other comment but perhaps we should revisit it and include it since it's [possible to run Windows on ARM64](https://docs.microsoft.com/en-us/windows/uwp/porting/apps-on-arm)?

Reply:
@pmm-sumo `otelcontribcol` currently does not compile with the following errors for windows on both `arm` and `arm64`:

```
# github.com/open-telemetry/opentelemetry-collector-contrib/pkg/winperfcounters/internal/third_party/telegraf/win_perf_counters
pkg/winperfcounters/internal/third_party/telegraf/win_perf_counters/pdh.go:345:89: undefined: PDH_FMT_COUNTERVALUE_DOUBLE
pkg/winperfcounters/internal/third_party/telegraf/win_perf_counters/performance_query.go:120:12: undefined: PDH_COUNTER_INFO
pkg/winperfcounters/internal/third_party/telegraf/win_perf_counters/performance_query.go:148:12: undefined: PDH_FMT_COUNTERVALUE_DOUBLE
pkg/winperfcounters/internal/third_party/telegraf/win_perf_counters/performance_query.go:171:15: [1 << 20]PDH_FMT_COUNTERVALUE_ITEM_DOUBLE used as value
pkg/winperfcounters/internal/third_party/telegraf/win_perf_counters/performance_query.go:171:24: undefined: PDH_FMT_COUNTERVALUE_ITEM_DOUBLE
```

I can open a follow up issue if we can move this pr forward, any thoughts?

Reply:
Thanks @pmm-sumo @mx-psi!